codegen/cmd/injection-gen/generators/reconciler_controller.go

```diff
@@ -212,11 +192,43 @@ func NewImpl(ctx {{.contextContext|raw}}, r Interface{{if .hasClass}}, classValu
 		if opts.FinalizerName != "" {
 			rec.finalizerName = opts.FinalizerName
 		}
+		if opts.AgentName != "" {
+			rec.Recorder = createRecorder(ctx, opts.AgentName)
```

Comment:
nit: I'd declare a string `agentName := defaultControllerAgentName` above the loop, set that here, and remove the if below.

Reply:
We still need if condition below because that condition checks if recorder is `nil`.